postman/Get Broadcast Open Clicks.postman_collection.json

```diff
@@ -0,0 +1,336 @@
+					"host": [
+						"{{base_url}}"
+					],
+					"path": [
+						"accounts"
+					]
+				}
+			},
+			"response": []
+		},
+		{
+			"name": "Get Lists on Account",
+			"event": [
+				{
+					"listen": "test",
+					"script": {
+						"id": "4d22c8fa-8695-4098-a275-10658e8aaea5",
+						"exec": [
+							"var jsonData = pm.response.json();",
+							"",
+							"pm.test(\"Retrieved more one or more lists for account\", function(){",
+							"    pm.expect(jsonData.total_size).is.be.above(0);",
+							"});",
+							"",
+							"pm.test(\"Successfully selected a random listId\", function() {",
+							"    var listId = Math.floor((Math.random() * jsonData.total_size ));",
+							"    if (typeof jsonData.entries[0].id == 'number' ) {",
+							"        pm.globals.set(\"listId\",jsonData.entries[0].id );",
+							"    }",
+							"});"
+						],
+						"type": "text/javascript"
+					}
+				}
+			],
+			"request": {
+				"method": "GET",
+				"header": [],
+				"body": {
+					"mode": "raw",
+					"raw": ""
+				},
+				"url": {
+					"raw": "{{base_url}}/accounts/{{accountId}}/lists",
+					"host": [
+						"{{base_url}}"
+					],
+					"path": [
+						"accounts",
+						"{{accountId}}",
+						"lists"
+					]
+				}
+			},
+			"response": []
+		},
+		{
+			"name": "Get Number of Broadcasts on List",
+			"event": [
+				{
+					"listen": "test",
+					"script": {
+						"id": "1bac9b8a-de19-4264-9c19-bdc1ea154a5c",
+						"exec": [
+							"pm.test(\"Successfully selected a random broadcast Id from list\", function() {",
+							"    var broadcast_entry = Math.floor((Math.random() * 100 ));",
+							"    if (typeof broadcast_entry == 'number' ) {",
+							"        pm.globals.set(\"broadcast_entry\", broadcast_entry );",
+							"    }",
+							"});"
+						],
+						"type": "text/javascript"
+					}
+				}
+			],
+			"request": {
+				"method": "GET",
+				"header": [],
+				"body": {
+					"mode": "raw",
+					"raw": ""
+				},
+				"url": {
+					"raw": "{{base_url}}/accounts/{{accountId}}/lists/{{listId}}/campaigns?campaign_type=b&ws.op=find&ws.show=total_size",
+					"host": [
+						"{{base_url}}"
+					],
+					"path": [
+						"accounts",
+						"{{accountId}}",
+						"lists",
+						"{{listId}}",
+						"campaigns"
+					],
+					"query": [
+						{
+							"key": "campaign_type",
+							"value": "b"
+						},
+						{
+							"key": "ws.op",
+							"value": "find"
+						},
+						{
+							"key": "ws.show",
+							"value": "total_size"
+						}
+					]
+				}
+			},
+			"response": []
+		},
+		{
+			"name": "Get Random Broadcast",
+			"event": [
+				{
+					"listen": "test",
+					"script": {
+						"id": "d2b1b30c-3aec-48cc-9f35-c1ab0683ee45",
+						"exec": [
+							"pm.test(\"Successfully retrieved a random broadcast\", function() {",
+							"    var jsonData = pm.response.json();",
+							"    pm.globals.set(\"broadcastId\", jsonData.entries[pm.globals.get(\"broadcast_entry\")].id);",
```

Comment:
This will only work if the list has at least 100 broadcasts. The only safe way to pick a random item from `entries` is to assert that there's at least 1 entry and use an index less than the size of `entries`.

Reply:
This also won't work if you pick a number between 0 and the total number of broadcasts on the list because that number could be above the 100 returned on a page.

You should pick a number less than the size of the entries array.